tests/unit/lms/models/module_item_configuration_test.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from lms.models import CanvasModuleItemConfiguration
+
+
+class TestCanvasModuleItemConfiguration:
+    def test_set_mapped_file_id_creates_a_new_mapping_if_none_exists(self, mic):
+        mic.set_mapped_file_id("original_file_id", "mapped_file_id")
+
+        assert mic.get_mapped_file_id("original_file_id") == "mapped_file_id"
+
+    def test_set_mapped_file_id_overwrites_an_existing_mapping_if_one_exists(self, mic):
+        mic.set_mapped_file_id("original_file_id", "mapped_file_id")
+
+        mic.set_mapped_file_id("original_file_id", "new_mapped_file_id")
+
+        assert mic.get_mapped_file_id("original_file_id") == "new_mapped_file_id"
```

Comment:
`set_mapped_file_id()` is written in such a way that SQLAlchemy's mutation tracking does work and changes do get persisted to the DB. This would be easy to get wrong since mutation tracking in nested dicts doesn't normally work even with `MutableDict`. This test is currently not covering that: the test would not fail if the mutation tracking and persistence were broken. But I'm not sure exactly how to write a test that _would_ fail.